geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisList.java

```diff
@@ -135,6 +161,27 @@ public int getDSFID() {
     return REDIS_LIST_ID;
   }
 
+  public int elementInsert(byte[] elementToInsert, byte[] referenceElement,
```

Comment:
I think this method also needs to be synchronized to protect against a concurrent GII which will serialize this RedisList instance. The GII does not do all the other locking that our redis ops does. So anytime we are making a structural change that is not safe with a concurrent "toData" call we need to add synchronization